packages/guides/integrating-the-safe-core-sdk.md

```diff
@@ -0,0 +1,353 @@
+
+```js
+import { ContractNetworksConfig } from '@gnosis.pm/safe-core-sdk'
+
+const id = await ethAdapter.getChainId()
+const contractNetworks: ContractNetworksConfig = {
+  [id]: {
+    multiSendAddress: '<MULTI_SEND_ADDRESS>',
+    safeMasterCopyAddress: '<MASTER_COPY_ADDRESS>',
+    safeProxyFactoryAddress: '<PROXY_FACTORY_ADDRESS>'
+  }
+}
+
+const safeFactory = await SafeFactory.create({ ethAdapter, contractNetworks })
+
+const safeSdk = await Safe.create({ ethAdapter, safeAddress, contractNetworks })
+```
+
+The `SafeFactory` constructor also accepts the property `safeVersion` to specify the Safe contract version that will deploy. This string can take the values `1.1.1`, `1.2.0` or `1.3.0` and if not specified the last version will be used by default.
+
+```js
+const safeVersion = 'X.Y.Z'
+const safeFactory = await SafeFactory.create({ ethAdapter, safeVersion })
+```
+
+## 3. Deploy a new Safe <a name="deploy-safe"></a>
+
+The Safe Core SDK library allows to deploy new Safes using the `safeFactory` instance we just created.
+
+Here, for example, we can create a new Safe account with 3 owners and 2 required signatures.
+
+```js
+import { SafeAccountConfig } from '@gnosis.pm/safe-core-sdk'
+
+const safeAccountConfig: SafeAccountConfig = {
+  owners: ['0x...', '0x...', '0x...']
+  threshold: 2,
+  // ... (optional params)
+}
+const safeSdk = await safeFactory.deploySafe(safeAccountConfig)
+```
+
+Calling the method `deploySafe` will deploy de desired Safe and return a Safe Core SDK initialized instance ready to be used.
+
+## 4. Create a transaction <a name="create-transaction"></a>
+
+The Safe Core SDK supports the execution of single Safe transactions but also MultiSend transactions. We can create a transaction object by calling the method `createTransaction` in our `Safe` instance.
+
+* **Create a single transaction**
+
+  This method can receive an object of type `SafeTransactionDataPartial` that represents the transaction we want to execute (once the signatures are collected). There are some optional properties in case we want to specify them.
+
+  ```js
+  import { SafeTransactionDataPartial } from '@gnosis.pm/safe-core-sdk-types'
+
+  const transaction: SafeTransactionDataPartial = {
+    to,
+    data,
+    value,
+    operation, // Optional
+    safeTxGas, // Optional
+    baseGas, // Optional
+    gasPrice, // Optional
+    gasToken, // Optional
+    refundReceiver, // Optional
+    nonce // Optional
+  }
+
+  const safeTransaction = await safeSdk.createTransaction(transaction)
+  ```
+
+* **Create a MultiSend transaction**
+
+  This method can receive an array of `MetaTransactionData` objects that represent the multiple transactions we want to include in our MultiSend transaction. If we want to specify some of the optional properties in our MultiSend transaction we can pass a second argument to the method `createTransaction` with the `SafeTransactionOptionalProps` object.
```

Comment:
"This method takes an array..."

Reply:
Not sure about the first comment.
This method takes this: 
```js
async createTransaction(
  safeTransactions: SafeTransactionDataPartial | MetaTransactionData[],
  options?: SafeTransactionDataPartial
): Promise<SafeTransaction>
```
In the the section **Create a single transaction** I used "This method can receive an object of type `SafeTransactionDataPartial`". But this method *can* also receive an array.

Reply:
"...can take..."